oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/token/JwtGenerator.java

```diff
@@ -125,7 +125,7 @@ public Jwt generate(OAuth2TokenContext context) {
 		}
 		// @formatter:on
 
-		JwsHeader.Builder headersBuilder = JwsHeader.with(SignatureAlgorithm.RS256);
+		JwsHeader.Builder headersBuilder = JwsHeader.with(registeredClient.getTokenSettings().getIdTokenSignatureAlgorithm());
```

Comment:
`registeredClient.getTokenSettings().getIdTokenSignatureAlgorithm()` should only be used when minting an ID Token. The existing default should be preserved for access tokens.